src/book/02-system/01-foundations/01-overview.mdx

```diff
@@ -22,3 +22,42 @@ After motions is hardware IO. The hardware will output the commands and input th
+
+The following is a table of roles that exist in the NUbots codebase and possible use cases for them.
+
+| Role Name            | What it contains                                                                                      | Usage                                                                                                                                                                                                                                                                   |
+| -------------------- | ----------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `balllocalisation`   | Real robot sensors, Visual Mesh, ball detector and ball localisation                                  | Used to test and tune how well the robot detects balls on a field.                                                                                                                                                                                                      |
+| `cameratest`         | Real robot sensors and camera                                                                         | Used to see if the cameras are working and often used for [calibrating the cameras](/guides/main/camera-calibration).                                                                                                                                                   |
+| `filter_playback`    | Filter sensor data, data playback and NUsight                                                         | Use for playing back raw sensor data and viewing the filtered data in NUsight.                                                                                                                                                                                          |
+| `firmwareinstaller`  | Firmware installer                                                                                    | Used to install firmware to the robot.                                                                                                                                                                                                                                  |
```

Comment:
This still says nothing about what this module does. What firmware? (hint: it is used to install firmware on the CM740)

Reply:
"says nothing" - that is completely true because I had no idea what it did 😆 Should have included a review comment asking about it but by the time I finished, I forgot 😵 
You say 'used to', do we not need it anymore?

Reply:
Read my comment again

Reply:
Ah, and with that I think it's time to sleep 😅